libraries/lts/include/mcrl2/lts/detail/liblts_bisim_m.h

```diff
@@ -0,0 +1,579 @@
+                to_be_processed.push_back((*i).block_index);
+            }
+        }
+
+        for (state_type i = 0; i < aut.num_states(); i++) 
+        {
+            block_index_type bid = block_index_of_a_state[i];
+            if (!block_flags[bid]) {
+                block_flags[bid] = true; 
+                partition.insert(bid);
+            }
+        }
+        mCRL2log(mcrl2::log::info) << "Partition refinement done, partition contains: " << partition.size() << " blocks, the history contains "<< lvl-1 << " levels." <<std::endl;
+        save_transitions();
+    }
+
+    /** \brief Destroys this partitioner. */
+    ~bisim_partitioner_counter_example() = default;
+
+    /** \brief Creates a state formula that distinguishes state s from state t.
+     *  \details The states s and t are non bisimilar states. A distinguishign state formula phi is
+     *           returned, which has the property that s \in \sem{phi} and  t \not\in\sem{phi}.
+     *           Based on the preprint "Computing minimal distinguishing Hennessey-Milner formulas is NP-hard
+     *           . But variants are tractable", 2023 by Jan Martens and Jan Friso Groote
+     *  \param[in] s The state number for which the resulting formula should be true
+     *  \param[in] t The state number for which the resulting formula should be false
+     *  \return A minimal observation depth distinguishing state formula, that is often also minimum negation-depth and irreducible. */
+    mcrl2::state_formulas::state_formula dist_formula_mindepth(const std::size_t s, const std::size_t t) {
+        formula f = distinguish(block_index_of_a_state[s], block_index_of_a_state[t]);
+        mCRL2log(mcrl2::log::info) << "done with formula \n";
+        return convertFormula(f);
+    };
+
+    bool in_same_class(const std::size_t s, const std::size_t t) 
+    {
+        return block_index_of_a_state[s] == block_index_of_a_state[t]; 
+    }
+
+
+private:
+    typedef std::size_t block_index_type;
+    typedef std::size_t state_type;
+    typedef std::size_t level_type;
+    typedef std::size_t formula_index_type;
+    typedef std::size_t label_type;
+    state_type initial_l2;
+
+    state_type max_state_index;
+    LTS_TYPE& aut;
+    std::set <block_index_type> partition;
+
+    struct block
+    {
+        state_type state_index;                   // The state number that represent the states in this block
+        block_index_type block_index;             // The sequence number of this block.
+        block_index_type parent_block_index;      // Index of the parent block.
+        level_type level;
+
+
+        // If there is no parent block, this refers to the block itself.
+        std::vector < state_type > states;
+        std::vector < transition > transitions;
+        std::set<std::pair < label_type, block_index_type >> outgoing_observations;
+
+        void swap(block& b)
+        {
+            state_type state_index1 = b.state_index;
```

Comment:
This can probably be refactored to std:;swap(b.state_index, state_index), same for the cases below.